atlasdb-impl-shared/src/main/java/com/palantir/atlasdb/transaction/service/TransactionServices.java

```diff
@@ -42,10 +42,10 @@ private static TransactionService createSplitKeyTransactionService(
             KeyValueService keyValueService,
             CoordinationService<InternalSchemaMetadata> coordinationService) {
         TransactionSchemaManager transactionSchemaManager = new TransactionSchemaManager(coordinationService);
+        int versionOne = 1;
         return new SplitKeyDelegatingTransactionService<>(
-                transactionSchemaManager::getTransactionsSchemaVersion,
-                ImmutableMap.of(1, createV1TransactionService(keyValueService))
-        );
+                new PreStartTimestampHandler<>(versionOne, transactionSchemaManager::getTransactionsSchemaVersion),
+                ImmutableMap.of(versionOne, createV1TransactionService(keyValueService)));
```

Comment:
- [ ] Add some explanation as to what is going on here / make this more descriptive